spot-contracts/contracts/RolloverVault.sol

```diff
@@ -394,6 +400,118 @@ contract RolloverVault is
         return redemptions;
     }
 
+    /// @notice Allows users to swap their underlying tokens for perps held by the vault
+    /// @param underlyingAmtIn The amount of underlying tokens swapped in.
+    /// @param execRecover If set to true, recovers vault assets to maximize liquidity for swapping.
+    /// @return The amount of perp tokens swapped out.
+    function swapUnderlyingForPerps(uint256 underlyingAmtIn, bool execRecover)
+        public
+        nonReentrant
+        whenNotPaused
+        returns (uint256)
+    {
+        // compute equal value perps to swap out
+        uint256 perpPrice = perp.getAvgPrice();
+        uint256 perpAmtOut = underlyingAmtIn.mulDiv(PERP_UNIT_PRICE, perpPrice);
+
+        // Compute fees
+        (uint256 swapFeePerpSharePerc, uint256 swapFeeVaultSharePerc) = feePolicy.computeUnderlyingToPerpSwapFeePercs(
+            underlyingAmtIn
+        );
+
+        // We compute perp's fee share
+        uint256 perpAmtToBurn = perpAmtOut.mulDiv(swapFeePerpSharePerc, FEE_ONE_PERC, MathUpgradeable.Rounding.Up);
```

Comment:
This is another tricky area with 1 vs 100 and decimals...

I'm always way more comfortable dividing by a number denominated in DECIMALS, rather than a scalar value. Because functionally, we're dividing in order to "fix decimals" not to affect the "numbers".

Commenting here, not because it's wrong per se, but it's definitely a code smell suggesting that we've organized our concepts incorrectly.

I forget now, are our decimals for 100 or for 1? Can we divide by decimals here without having to, e.g. subtract 2 from it?

Reply:
FEE_ONE_PERC = 100% or 1.0